src/util.rs

```diff
@@ -5,6 +5,14 @@
 
 use crate::field::{FieldElement, FieldError};
 
+/// Possible errors from proofs
+#[derive(Debug, thiserror::Error)]
+pub enum ProofError {
+    /// Input sizes do not match
+    #[error("input sizes do not match")]
+    InputSizeMismatch,
+}
```

Comment:
The name "ProofError" sound as if this might be returned if the input-validation protocol failed. Looking at the error that's propagated, this seems to happen only when "unpacking" a proof. Perhaps this is better cast as another "SerializeError" type error? Something like "SerializeError::UnpackInputSizeMismatch"?

Reply:
That's a good point. The tricky thing is that I want to make a follow up PR where I move this enum, `struct UnpackedProof`, `struct UnpackedProofMut`, `unpack_proof` and `unpack_proof_mut` to a new module `proof.rs` (the idea is to remove stuff from `util.rs` with the eventual goal of deleting that module). But arguably, all the stuff here in `util.rs` has to do with serialization, so maybe instead of `proof.rs` my later PR should just rename `util.rs` to `serialize.rs`...

OK, given that we want to revisit serialization in the near future anyway (#24), I think I'm just going to punt on moving anything from `util.rs` into a new module, and for now I will add a case to `SerializeError` as you suggest.

Reply:
IMO `serialize` is a better name for the stuff in this package than `proof`.